uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/leave-request-popup-controller_test.js

```diff
@@ -0,0 +1,591 @@
+
+          it('selects by default', function () {
+            expect($ctrl.uiOptions.multipleDays).toBeTruthy();
+          });
+
+        });
+
+        describe('single', function () {
+
+          beforeEach(function () {
+            $ctrl.uiOptions.multipleDays = false;
+            $ctrl.changeInNoOfDays();
+            $scope.$digest();
+          });
+
+          it('selects single day', function () {
+            expect($ctrl.uiOptions.multipleDays).toBeFalsy();
+          });
+
+          it('hides to date and type', function () {
+            expect($ctrl.uiOptions.toDate).not.toBeDefined();
+            expect($ctrl.uiOptions.selectedToType).not.toBeDefined();
+          });
+
+          it('will reset balance and types', function () {
+            expect($ctrl.uiOptions.selectedFromType).not.toBeDefined();
+            expect($ctrl.uiOptions.selectedToType).not.toBeDefined();
+            expect($ctrl.balance.closing).toEqual(0);
+            expect($ctrl.balance.change.amount).toEqual(0);
+          });
+        });
+      });
+
+      describe('calendar', function () {
+
+        it('no dates are selected by default', function () {
+          expect($ctrl.uiOptions.fromDate).not.toBeDefined();
+          expect($ctrl.uiOptions.toDate).not.toBeDefined();
+        });
+
+        describe('from date', function () {
+
+          beforeEach(function () {
+            $ctrl.onDateChange(new Date(), true);
+            $scope.$digest();
+          });
+
+          it('can change date', function () {
+            expect(moment($ctrl.leaveRequest.from_date, serverDateFormat, true).isValid()).toBe(true);
+          });
+
+        });
+
+        describe('to date', function () {
+
+          beforeEach(function () {
+            $ctrl.onDateChange(new Date(), false);
+            $scope.$digest();
+          });
+
+          it('can change date', function () {
+            expect(moment($ctrl.leaveRequest.to_date, serverDateFormat, true).isValid()).toBe(true);
+          });
+
+        });
+      });
+
+      describe('day types', function () {
+
+        it('by default no selection', function () {
+          expect($ctrl.uiOptions.selectedFromType).not.toBeDefined();
+          expect($ctrl.uiOptions.selectedToType).not.toBeDefined();
+        });
+
+        describe('from', function () {
+
+          beforeEach(function () {
+            spyOn($ctrl, 'filterLeaveRequestDayTypes').and.callThrough();
+            $ctrl.onDateChange(new Date(), true);
+            $scope.$digest();
+          });
+
+          it('by default selects first day type', function () {
+            expect($ctrl.uiOptions.selectedFromType).toEqual($ctrl.leaveRequestFromDayTypes[0]);
+          });
+
+          it('will filter based on current date', function () {
+            expect($ctrl.filterLeaveRequestDayTypes).toHaveBeenCalledWith(jasmine.any(Date), jasmine.any(Boolean));
+          });
+
+          describe('is public holiday', function () {
```

Comment:
I'm not sure what you're testing here? You are checking what? In case the "from date" is a public holiday? What about the other day types?